src/main/java/com/powsybl/openloadflow/network/LfAction.java

```diff
@@ -34,7 +31,7 @@ public final class LfAction {
     private record TapPositionChange(LfBranch branch, int value, boolean isRelative) {
     }
 
-    private record LoadShift(String loadId, LfLoad load, PowerShift powerShift) {
+    private record LoadShift(Load load, LfLoad lfLoad, PowerShift powerShift) {
     }
```

Comment:
You should not keep iidm.network objects in memory since fast restart is implemented (https://github.com/powsybl/powsybl-open-loadflow/pull/635)
If you need to keep the Load in a record, use a WeakRef<Load> instead.
Otherwise, this will prevent the GC from deleting a Network if the fast restart mode is on.